### PR TITLE
Filter autotask runs by status

### PR DESCRIPTION
**Pull request by zeljkoX**
This PR will add support for filtering autotask runs by status.

See: https://github.com/OpenZeppelin/defender/pull/3476

### CODE REVIEW

File: packages/autotask-client/src/api.ts [hunk 1]
```diff
@@ -85,9 +90,17 @@ export class AutotaskClient extends BaseApiClient {
     return this.updateCode(autotaskId, encodedZippedCode);
   }
 
-  public async listAutotaskRuns(autotaskId: string, next?: string): Promise<AutotaskRunListResponse> {
+  public async listAutotaskRuns(
+    autotaskId: string,
+    next?: string,
+    status?: AutotaskRunStatus | undefined,
+  ): Promise<AutotaskRunListResponse> {
+    if (next && !status && (next === 'success' || next === 'error' || next === 'pending' || next === 'throttle')) {
```
Comment (mok0230) [inline packages/autotask-client/src/api.ts:98]: I understand this approach in the context of not introducing a breaking change for something this minor, but this feels worthy of noting in a "Next time we make breaking changes" document. Do you know if we have a document like this now? If not, we could make a new "Defender Client v1 => v2 Breaking Changes" Project in Linear and add this.

**Review by mok0230 — APPROVED** [review]

File: packages/autotask-client/src/api.ts [hunk 2]
```diff
@@ -85,9 +90,17 @@ export class AutotaskClient extends BaseApiClient {
     return this.updateCode(autotaskId, encodedZippedCode);
   }
 
-  public async listAutotaskRuns(autotaskId: string, next?: string): Promise<AutotaskRunListResponse> {
+  public async listAutotaskRuns(
+    autotaskId: string,
+    next?: string,
```
Comment (BjMrq) [inline packages/autotask-client/src/api.ts:95]: Wondering what "next" is? Should it be typed AutotaskRunStatus here?

maybe we could combine `next === 'success' || next === 'error' || next === 'pending' || next === 'throttle')` and `next as AutotaskRunStatus;` as a function using typescript "is" ?

**Review by BjMrq — APPROVED** [review]

File: packages/autotask-client/src/api.ts (hunk 2 above)
Comment (zeljkoX) [reply packages/autotask-client/src/api.ts:95]: `next` is dynamo paging cursor in this context.

File: packages/autotask-client/src/api.ts (hunk 1 above)
Comment (zeljkoX) [reply packages/autotask-client/src/api.ts:98]: We probably don't have something like that.

In general we should strive to use object params as arguments as they are easier to extend. No breaking changes that way.

@dylankilkenny FYI as you are working on new client library that could we wrapper for already existing ones also.

Create Project/issue: https://linear.app/libraries-and-tooling/issue/LIB-561/defender-client-v1-candidates-for-breaking-changes

**Review by pasevin — APPROVED** [review]

File: packages/autotask-client/src/api.ts (hunk 1 above)
Comment (mok0230) [reply packages/autotask-client/src/api.ts:98]: > In general we should strive to use object params as arguments as they are easier to extend.

Agreed!

> @dylankilkenny FYI as you are working on new client library that could we wrapper for already existing ones also.

+1...may be worth identifying any other tech debt/breaking change items like this where we may want to improve the API but would have required a breaking change in `defender-client`. If anyone has other items, please add to the project @zeljkoX  just created: [https://linear.app/libraries-and-tooling/issue/LIB-561](https://linear.app/libraries-and-tooling/issue/LIB-561/defender-client-v1-candidates-for-breaking-changes)

**Pull request merged**